_posts/2021-05-06-broker-tuning.md

```diff
@@ -0,0 +1,454 @@
+Base the value on the average message size and throughput. When considering the total memory allocation required for read/write buffering, the memory available must also be able to accommodate the maximum replicated message size when multiplied by all followers. The size must also be greater than `message.max.bytes`, otherwise the broker can accept messages that are too big to replicate.
+
+```properties
+# ...
+replica.fetch.max.bytes=1048576
+# ...
+```
+
+The importance of Kafka's topic replication mechanism cannot be overstated.
+Topic replication is central to Kafka's reliability and data durability. Using replication, a failed broker can recover from in-sync replicas on other brokers.
+We go into more detail about leaders, followers and in-sync replicas with [partition rebalancing for availability](#partition-rebalancing-for-availability).
+
+#### Creating and deleting topics
+
+The `auto.create.topics.enable` property is enabled by default so that topics are created when needed by a producer or consumer. It's usually disabled in production as Kafka users tend to prefer applying more control over topic creation in that environment.
+If you are using automatic topic creation, you can set the default number of partitions for topics using `num.partitions`. Use it with `default.replication.factor` property. In this case, you might want to set the replication factor to the number of brokers or to at least three replicas.
+The `delete.topic.enable` property is enabled by default to allow topics to be deleted. Kafka users normally disable this property in production too. This time you're making sure that data is not accidentally lost, although you can temporarily enable the property to delete topics if circumstances demand it.
+
+```properties
+# ...
+auto.create.topics.enable=false
+delete.topic.enable=true
+# ...
+```
+
+> You cannot delete topics with the `KafkaTopic` resource if the `delete.topic.enable` property is set to false.
+
+### Internal topic settings for transactions and commits
+
+Pay attention to the configuration requirements for internal Kafka topics.
+
+If you are [using transactions](https://strimzi.io/docs/operators/latest/using.html#reliability_guarantees) to enable atomic writes to partitions from producers, the internal `__transaction_state` topic used in the process requires a minimum of three brokers with the default settings.
+
+```properties
+# ...
+transaction.state.log.replication.factor=3
+transaction.state.log.min.isr=2
+# ...
+```
+
+The `__consumer_offsets` topic, which stores partition offset commits, has default settings for the number of partitions and replication factor.
+
+```properties
+# ...
+offsets.topic.num.partitions=50
+offsets.topic.replication.factor=3
+offsets.commit.required.acks=-1
+# ...
+```
+
+***:warning: Do not reduce these settings in production.***
```

Comment:
This does not seem to render on the website ... it just shows `:warning: Do not reduce these settings in production.` in bold. Maybe we can do something like:

```suggestion
***WARNING: Do not reduce these settings in production.***
```